looprpc/server.proto

```diff
@@ -80,6 +88,9 @@ message ServerLoopInRequest {
     uint64 amt = 3;
     string swap_invoice = 4;
     bytes last_hop = 5;
+
+    /// The protocol version that the client adheres to.
+    ProtocolVersion protocol_version = 6;
```

Comment:
Question is, should we version all RPC calls? Or just loops?
Should we use server reflection instead? (https://github.com/grpc/grpc-go/tree/master/examples/features/reflection)

Reply:
Can you get a version number with server reflection?

What is versioned here is just the protocol version (scripts etc), not the RPC interface. For that, we'd need to have a separate endpoint like `GetVersion` in `lnd`.

Reply:
Going to look into the grpc context, maybe that can carry the version number.

Reply:
Discussed with @guggero and grpc metadata doesn't seem a great idea. It is all string based (no nice enum) and also proxies may change the keys :fearful:

Reply:
Added version to all RPC calls.

Reply:
Server reflection may be used for protocol discovery, which would be useful for clients to check if their protocol description is compatible with the server protocol description. Afaik context may be used to store a version number (it's similar to http headers) but that also is somewhat hidden (as in not reflected in the proto definitions).

Reply:
The point is that not every aspect of the protocol (which for example also includes the on-chain htlc bitcoin script), can be observed from the proto signatures.

Using the grpc metadata in the context seems not very attractive (see comment above)